modules/consul/consul.go

```diff
@@ -0,0 +1,76 @@
+package consul
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const (
+	defaultHttpApiPort = "8500"
+	defaultBrokerPort  = "8600"
+)
+
+const (
+	DefaultBaseImage = "docker.io/hashicorp/consul:1.15"
+)
+
+// ConsulContainer represents the Consul container type used in the module.
+type ConsulContainer struct {
+	testcontainers.Container
+}
+
+// ApiEndpoint returns host:port for the HTTP API endpoint.
+func (c *ConsulContainer) ApiEndpoint(ctx context.Context) (string, error) {
+	mappedPort, err := c.MappedPort(ctx, defaultHttpApiPort)
+	if err != nil {
+		return "", err
+	}
+
+	hostIP, err := c.Host(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	uri := fmt.Sprintf("%s:%s", hostIP, mappedPort.Port())
+	return uri, nil
+}
+
+// WithLocalConfig passes a JSON string of keys and values to define a configuration to be used by the instance.
+func WithLocalConfig(config string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		req.Env["CONSUL_LOCAL_CONFIG"] = config
```

Comment:
Not a blocker, just a question: do we want to accept a simple string or an io.Reader to allow reading configs from file?

Reply:
I like the idea of having an io.Reader here. Perhaps I should add another Option that implements it? I wouldn't be too sure about replacing the string parameter altogether as it might be useful to test a certain parameters more quickly. What would you say?

Reply:
We could have a `WithLocalConfigReader(r io.Reader)` as a generalisation of the read process, keeping the string specialisation.

Reply:
Another option might be something like `WithConfigFile(path string)`, and then map the file to the container. Would you say that's preferred?

Reply:
We usually have that in database modules for passing config files, so I think it'd make sense to add it